rfcs/000-remove-double-relationships.md

```diff
@@ -0,0 +1,102 @@
+- Feature Name: Remove double relationships between Neo4j nodes
+- Start Date: 2021-05-25
+- RFC PR: [amundsen-io/rfcs#0000](https://github.com/amundsen-io/rfcs/pull/0000)
+- Amundsen Issue: 
+
+# Remove double relationships between Neo4j nodes
+
+## Summary
+
+Currently all (most?) relationships between Neo4j nodes are modeled with
+two directional edges. This RFC proposes we only use one directional edge
+between the nodes.
+
+## Motivation
+
+The double relationship graph model makes Neo4j queries more complicated
+than they need to be, with no performance or functional benefit. Additionally
+due to conflicting changes in the past (PR 70 and PR 206), Amundsen instances
+that have been running and ingesting data since prior to PR 70 or PR 206 have
+inconsistent Neo4j database state. Some nodes have a single relationship between
+then, some have two.
+
+There are two relevant Slack discussions regarding this change:
+
+https://app.slack.com/client/TGFR0CZM3/CGFBVT23V/thread/CGFBVT23V-1615391140.022900
+- a thread where I asked the question on why the double relationships exist
+https://app.slack.com/client/TGFR0CZM3/CGFBVT23V/thread/CGFBVT23V-1563487877.254000
+- a thread with Joshua Haskins' benchmarking results for modeling the relationships
+  with one or two edges
+
+## Transition Path
+
+The implementation work involved:
+
+For each double relationship pair, we need to decide which relationship type
```

Comment:
Are you thinking there is a way to apply this generically? Perhaps asked more broadly - what else may need to be done by companies that have created their own models?

Reply:
Good point. I wasn't thinking of custom models at all. I'm not sure what the right approach would be for those.

Maybe offer guidance on how to make corresponding changes on their own code, if needed? Then leave it at their own discretion whether they want to do the change.

Reply:
My 2CS here is that if you’re deviating from the standard datamodels (although I *think* it’s only an issue if you modify, rather than add) then that’s kind of you. What we can do to help is create some very clear documentation of what the data models are, rather than having to extract them from source code.